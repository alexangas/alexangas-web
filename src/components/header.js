import React from "react"
import { useStaticQuery, graphql, Link } from "gatsby"

export const PureHeader = ({ location, data }) => {
  const { title } = data.site.siteMetadata

  const rootPath = `${__PATH_PREFIX__}/`
  const blogPath = `${__PATH_PREFIX__}/blog/`
  let header

  if (location.pathname === rootPath || location.pathname === blogPath) {
    header = (
      <h1 className="title">
        <Link to={`/`} className="has-text-dark">
          {title}
        </Link>
      </h1>
    )
  } else {
    header = (
      <h3 className="subtitle">
        <Link to={`/blog/`}>{title}</Link>
      </h3>
    )
  }

  return <header>{header}</header>
}

export const Header = (props) => {
  const data = useStaticQuery(graphql`
    query HeaderQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  return <PureHeader {...props} data={data} />
}

export default Header
