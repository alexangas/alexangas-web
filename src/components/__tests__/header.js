import React from "react"
import renderer from "react-test-renderer"

import { PureHeader as Header } from "../header"

describe("Header", () => {
  it("renders root path correctly", () => {
    const data = {
      site: {
        siteMetadata: {
          title: "Name of Blog",
        },
      },
    }
    const location = {
      pathname: "/",
    }
    const testRenderer = renderer.create(
      <Header location={location} data={data} />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders blog path correctly", () => {
    const data = {
      site: {
        siteMetadata: {
          title: "Name of Blog",
        },
      },
    }
    const location = {
      pathname: "/blog/",
    }
    const testRenderer = renderer.create(
      <Header location={location} data={data} />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders non-blog path correctly", () => {
    const data = {
      site: {
        siteMetadata: {
          title: "Some Page",
        },
      },
    }
    const location = {
      pathname: "/some-page/",
    }
    const testRenderer = renderer.create(
      <Header location={location} data={data} />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
