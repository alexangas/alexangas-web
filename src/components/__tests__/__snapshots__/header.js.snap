// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders blog path correctly 1`] = `
<header>
  <h1
    style={
      Object {
        "fontSize": "2.05811rem",
        "lineHeight": "2.415rem",
      }
    }
  >
    <a
      href="/"
    >
      Name of Blog
    </a>
  </h1>
</header>
`;

exports[`Header renders non-blog path correctly 1`] = `
<header>
  <h3>
    <a
      href="/blog/"
    >
      Some Page
    </a>
  </h3>
</header>
`;

exports[`Header renders root path correctly 1`] = `
<header>
  <h1
    style={
      Object {
        "fontSize": "2.05811rem",
        "lineHeight": "2.415rem",
      }
    }
  >
    <a
      href="/"
    >
      Name of Blog
    </a>
  </h1>
</header>
`;
